src/App.tsx

```diff
@@ -1,61 +1,183 @@
+import { useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
 
 export const App = () => {
+  const [todoTitle, setTodoTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState(0);
+
+  const [hasUserError, setHasUserError] = useState(false);
+  const [hasTodoTitleError, setHasTodoTitleError] = useState(false);
+
+  const userNotFound = {
+    id: 0,
+    name: 'user',
+    email: 'user@mail.com',
+  };
+
+  const getTodosWithUsers = () => {
+    return todosFromServer.map((todo) => {
+      const user = usersFromServer.find(
+        ({ id }) => id === todo.userId,
+      ) || userNotFound;
+      const { name, id } = user;
+
+      return {
+        ...todo,
+        name,
+        userId: id,
+        user,
+      };
+    });
+  };
+
+  const defaultTodosWithUsers = getTodosWithUsers();
+
+  const [todosWithUsers, setTodosWithUsers]
+    = useState<Todo[]>(defaultTodosWithUsers);
+
+  const getMaxTodoId = ():number => {
+    return Math.max(...todosWithUsers.map(({ id }) => id));
+  };
+
```

Comment:
Move this function out of component, don't forget that functions that are created inside component - will be created on every component render and in most cases we don't want it